src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -55,9 +55,20 @@ private[avro] class AvroRelation(
   private val recordNamespace = parameters.getOrElse("recordNamespace", "")
 
   /** needs to be lazy so it is not evaluated when saving since no schema exists at that location */
-  private lazy val avroSchema = paths match {
-    case Array(head, _*) => newReader(head)(_.getSchema)
-    case Array() => throw NoFilesException
+  private lazy val avroSchema = if (paths.isEmpty) {
+    throw NoFilesException
+  } else {
+    // As of Spark 1.5.0, it's possible to receive an array which contains a single non-existent
```

Comment:
@liancheng, could you take a look at this change? It looks like the `globPathIfNecessary` change in Spark 1.5.0 means that non-existent paths may get passed down to data sources if those paths didn't contain any glob characters. In the cases where this happens, though, I think that we will only receive an array with a single path, so the third-party data source code should only need to check path existence when `paths.size == 1`. Would be good to confirm this intuition, though.